pkg/blackduck/ctl_blackduck.go

```diff
@@ -235,171 +227,254 @@ func FlagWasSet(flagset *pflag.FlagSet, flagName string) bool {
 	return false
 }
 
-// AddHelmValueByCobraFlag updates a field in the blackDuckSpec if the flag was set by the user. It gets the
-// value from the corresponding struct field.
-// Note: It should only handle values with a 1 to 1 mapping - struct-field to spec
-func (ctl *HelmValuesFromCobraFlags) AddHelmValueByCobraFlag(f *pflag.Flag) {
-	if f.Changed {
-		log.Debugf("flag '%s': CHANGED", f.Name)
-		switch f.Name {
-		case "version":
-			util.SetHelmValueInMap(ctl.args, []string{"imageTag"}, ctl.flagTree.Version)
-		case "size":
-			util.SetHelmValueInMap(ctl.args, []string{"size"}, strings.ToLower(ctl.flagTree.Size))
-		case "expose-ui":
-			util.SetHelmValueInMap(ctl.args, []string{"exposeui"}, true)
-			switch ctl.flagTree.ExposeService {
-			case util.NODEPORT:
-				util.SetHelmValueInMap(ctl.args, []string{"exposedServiceType"}, "NodePort")
-			case util.LOADBALANCER:
-				util.SetHelmValueInMap(ctl.args, []string{"exposedServiceType"}, "LoadBalancer")
-			case util.OPENSHIFT:
-				util.SetHelmValueInMap(ctl.args, []string{"exposedServiceType"}, "OpenShift")
-			default:
-				util.SetHelmValueInMap(ctl.args, []string{"exposeui"}, false)
-			}
-		case "environs":
-			for _, value := range ctl.flagTree.Environs {
-				values := strings.SplitN(value, ":", 2)
-				if len(values) != 2 {
-					panic(fmt.Errorf("invalid environ configuration for %s", value))
+// GenerateHelmFlagsFromCobraFlags checks each flag in synopsysctl and updates the map to
+// contain the corresponding helm chart field and value
+func (ctl *HelmValuesFromCobraFlags) GenerateHelmFlagsFromCobraFlags(flagset *pflag.FlagSet) (map[string]interface{}, error) {
+	err := ctl.CheckValuesFromFlags(flagset)
+	if err != nil {
+		return nil, err
+	}
+	flagset.VisitAll(func(f *pflag.Flag) {
```

Comment:
The purpose ignoring the method to add the function directly is to report all errors at once instead of them reporting individually. Currently we are logging fatal exception which might not solve that use case.

Reply:
I made it like this so that the cases would have access to the flagset instead of just individual flags... but I switched all the Fatalf() instances to Errorf()  :)